### PR TITLE
fix: broken draw io diagram on rtl mode, modify elastic search config

### PR DESCRIPTION
**Pull request by avioral**
# Diagram 
- Fix broken draw io diagram on rtl mode

# Elastic search
- Improve query
- Add support for tags
- Add support for fields boosting
- Reject empty inputs on build suggest

### CODE REVIEW

File: server/modules/search/elasticsearch/engine.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = {
     WIKI.logger.info(`(SEARCH/ELASTICSEARCH) Initializing...`)
     switch (this.config.apiVersion) {
       case '7.x':
-        const { Client: Client7 } = require('elasticsearch7')
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:22]: Restore the spaces in objects to respect the project formatting rules.

File: server/modules/search/elasticsearch/engine.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ module.exports = {
         })
         break
       case '6.x':
-        const { Client: Client6 } = require('elasticsearch6')
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:31]: Restore the spaces in objects to respect the project formatting rules.

File: server/modules/search/elasticsearch/engine.js [hunk 3]
```diff
@@ -50,18 +50,19 @@ module.exports = {
    */
   async createIndex() {
     try {
-      const indexExists = await this.client.indices.exists({ index: this.config.indexName })
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:53]: Restore the spaces in objects to respect the project formatting rules.

File: server/modules/search/elasticsearch/engine.js [hunk 4]
```diff
@@ -50,18 +50,19 @@ module.exports = {
...
-              content: { type: 'text', boost: 1.0 },
-              locale: { type: 'keyword' },
-              path: { type: 'text' }
+              suggest: {type: 'completion'},
+              title: {type: 'text', boost: 10.0},
+              description: {type: 'text', boost: 3.0},
+              content: {type: 'text', boost: 1.0},
+              locale: {type: 'keyword'},
+              path: {type: 'text'},
+              tags: {type: 'text', boost: 8.0}
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:65]: Restore the spaces in objects to respect the project formatting rules.

File: server/modules/search/elasticsearch/engine.js [hunk 5]
```diff
@@ -92,30 +93,16 @@ module.exports = {
...
+            simple_query_string: {
+              query: `*${q}*`,
+              fields: ['title^20', 'description^3', 'tags^8', 'content^1'],
+              default_operator: 'and',
+              analyze_wildcard: true
             }
           },
           from: 0,
           size: 50,
+          explain: true,
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:105]: Necessary in non-dev?

File: server/modules/search/elasticsearch/engine.js [hunk 6]
```diff
@@ -251,7 +253,7 @@ module.exports = {
    */
   async rebuild() {
     WIKI.logger.info(`(SEARCH/ELASTICSEARCH) Rebuilding Index...`)
-    await this.client.indices.delete({ index: this.config.indexName })
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:254]: Restore the spaces in objects to respect the project formatting rules.

File: server/modules/search/elasticsearch/engine.js [hunk 7]
```diff
@@ -324,8 +328,9 @@ module.exports = {
       bytes = 0
     }
 
+    // Added real id in order to fetch page tags from the query
     await pipeline(
-      WIKI.models.knex.column({ id: 'hash' }, 'path', { locale: 'localeCode' }, 'title', 'description', 'render').select().from('pages').where({
```
Comment (NGPixel) [inline server/modules/search/elasticsearch/engine.js:328]: Restore the spaces in objects to respect the project formatting rules.

**Review by NGPixel — CHANGES_REQUESTED** [review]

File: server/modules/search/elasticsearch/engine.js (hunk 5 above)
Comment (avioral) [reply server/modules/search/elasticsearch/engine.js:105]: It is indeed not necessary i will omit this line

**Review by NGPixel — APPROVED** [review]

**Pull request merged**